systems/plants/setupLCPmex.cpp

```diff
@@ -0,0 +1,275 @@
+inline void partitionMatrix(vector<bool> const &indices, MatrixXd const & M, MatrixXd & included, MatrixXd & excluded)
+{   
+  const int n = indices.size();
+  const int count = getNumTrue(indices);
+  const int cols = M.cols();
+
+  included = MatrixXd::Zero(count, cols);
+  excluded = MatrixXd::Zero(n-count, cols);
+
+  int inclusionIndex = 0;
+  int exclusionIndex = 0;
+
+  for (int x = 0; x < n; x++) {
+    if (indices[x]) {
+      included.row(inclusionIndex++) = M.row(x);
+    } else {
+      excluded.row(exclusionIndex++) = M.row(x);
+    }
+  }
+}
+
+//builds a filtered matrix containing only the rows specified by indices
+inline void filterByIndices(vector<int> const &indices, MatrixXd const & M, MatrixXd & filtered)
+{
+  const int n = indices.size();
+  filtered = MatrixXd::Zero(n, M.cols());
+  for (int x = 0; x < n; x++) {
+    filtered.row(x) = M.row(indices[x]);
+  }
+}
+
+//[M, w, Mqdn, wqdn] = setupLCPmex(mex_model_ptr, q, qd, u, phiC, n, D, h, z_inactive_guess_tol)
+void mexFunction( int nlhs, mxArray *plhs[],int nrhs, const mxArray *prhs[] ) { 
+  RigidBodyManipulator *model= (RigidBodyManipulator*) getDrakeMexPointer(prhs[0]);
+  const int nq = model->num_positions;
+  const int nv = model->num_velocities;
+  const int numContactPairs = mxGetNumberOfElements(prhs[4]);
+  const int mC = mxGetNumberOfElements(prhs[6]);
+  const double z_inactive_guess_tol = static_cast<double>(mxGetScalar(prhs[8]));
+  const double h = static_cast<double>(mxGetScalar(prhs[7]));
+  const Map<VectorXd> q(mxGetPr(prhs[1]),nq);
+  const Map<VectorXd> v(mxGetPr(prhs[2]),nv);
+  const Map<VectorXd> u(mxGetPr(prhs[3]),model->B.cols());
+  
+  const Map<VectorXd> phiC(mxGetPr(prhs[4]),numContactPairs);
+  const Map<MatrixXd> n(mxGetPr(prhs[5]), numContactPairs, nq);
+
+  VectorXd C, phiL, phiP, phiL_possible, phiC_possible, phiL_check, phiC_check;
+  MatrixXd H, B, JP, JL, JL_possible, n_possible, JL_check, n_check;
+
+  manipulatorDynamics(model, q, v, H, C, B);
+
+  model->positionConstraints(phiP, JP);
+  model->jointLimitConstraints(q, phiL, JL);
+  MatrixXd Hinv = H.inverse();
+  const int nP = phiP.size();
+  
+  plhs[2] = mxCreateDoubleMatrix(nq, 1, mxREAL);
+  Map<VectorXd> wqdn(mxGetPr(plhs[2]), nq);
+  wqdn = v + h*Hinv*(B*u - C);
+
+  //use forward euler step in joint space as
+  //initial guess for active constraints
+  vector<bool> possible_contact;
+  vector<bool> possible_jointlimit;
+  getThresholdInclusion(phiC + h*n*v, z_inactive_guess_tol, possible_contact);
+  getThresholdInclusion(phiL + h*JL*v, z_inactive_guess_tol, possible_jointlimit);
+
+  while (true) {
+  //continue from here if our inactive guess fails
+    const int nC = getNumTrue(possible_contact);
+    const int nL = getNumTrue(possible_jointlimit);
+    const int LCP_size = nL+nP+(mC+2)*nC;
+   
+    plhs[0] = mxCreateDoubleMatrix(LCP_size, 1, mxREAL);
+    plhs[1] = mxCreateDoubleMatrix(nq, LCP_size, mxREAL);
```

Comment:
They have to be since they depend on LCP_size which can change at every iteration.  I should probably delete them first before overwriting the old one, though.  Fixing it.